README.md
MantaRay Forensics
==================================
ManTech Triage and Analysis System, Forensics Workflow Automation Suite

![alt tag](https://raw.githubusercontent.com/mantarayforensics/mantaray/master/mantaray/images/Mantaray_Logo_Template_Full_Screen.gif)

Overview
==================================

MantaRay is designed to automate the processing of forensic images, 
directories and individual files with open source tools. With 
support for numerous image formats, this tool provides a scalable 
base to utilize open source and custom exploitation tools.

For more information about the suite, visit our website http://www.mantarayforensics.com

Dependencies
==================================

See https://launchpad.net/~mantaray/+archive/stable for a full list

* BulkExtractor
* ENT – Calculate Entropy
* KML from JPG EXIF Data
* fdupes
* EXIF Tool
* Foremost
* Jumplist Parser
* Sleuth Kit tools
* Regripper
* Log2Timeline
* Volatility

Installation of MantaRay
==================================

Using bootstrap to install Mantaray:

    wget -q https://raw.githubusercontent.com/mantarayforensics/mantaray/master/mrbootstrap.sh && sh mrbootstrap.sh -i

Using bootstrap to install Mantaray, and the MantaRay Skin:

    wget -q https://raw.githubusercontent.com/mantarayforensics/mantaray/master/mrbootstrap.sh && sh mrbootstrap.sh -s -i


####PPA:

    sudo apt-add-repository ppa:mantaray/stable
    sudo apt-add-repository ppa:sift/stable
    sudo apt-get update && sudo apt-get upgrade -y
    sudo apt-get install mantaray
        
####Source from PPA:

    sudo apt-add-repository ppa:mantaray/stable
    apt-get source mantaray
    
####GitHub:

    git clone https://github.com/mantarayforensics/mantaray.git
    cd mantaray
    
Follow PPA directions to install dependencies

How To Run MantaRay
==================================
From command line (as user with sudo privileges) in PPA:

    sudo mantaray


A popup window should appear. Pressing continue will begin running the script.

Check For Updates
==================================
Using apt-get:

    sudo apt-get update && sudo apt-get upgrade

Using mantaray-updater:

    sudo mantaray-updater
    
GitHub:

    git pull origin master

Errors and Bugs
==================================
If MantaRay crashes, please re-run it in debug mode and send a screenshot
of the crash along with any other details you can provide to the 
MantaRay Forum http://mantarayforensics.com/forums/
or report to our GitHub https://github.com/mantarayforensics/mantaray/issues


debian/changelog
mantaray (1.3.92-ubuntu03) precise; urgency=low

  * Corrected fn dependency issue

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Mon, 10 Nov 2014 13:40:50 -0800

mantaray (1.3.92-ubuntu02) precise; urgency=low

  * updates pre-v1.4 release

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Mon, 10 Nov 2014 13:15:59 -0800

mantaray (1.3.92-ubuntu01) precise; urgency=low

  * Corrected issue with removed file

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Wed, 29 Oct 2014 15:24:14 -0700

mantaray (1.3.92-ubuntu00) precise; urgency=low

  * Prep for v1.4 update

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Wed, 29 Oct 2014 14:16:11 -0700

mantaray (1.3.91-ubuntu40) precise; urgency=low

  * Corrected dependencies for Volatility Update

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Wed, 03 Sep 2014 09:12:50 -0400

mantaray (1.3.91-ubuntu3) precise; urgency=low

  * Removed dependency of SIFT

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Wed, 27 Aug 2014 08:01:08 -0400

mantaray (1.3.91-ubuntu0) precise; urgency=low

  * Corrections to support Trusty build.

 -- Chapin Bryce (launchpad) <webmaster@mantarayforensics.com>  Sun, 06 Jul 2014 12:51:41 -0700

mantaray (1.3.90-ubuntu0) precise; urgency=low

  * Release of 1.3.9

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Tue, 03 Jun 2014 18:47:35 -0400

mantaray (1.3.83-ubuntu0) precise; urgency=low

  * Fixed Launcher error

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Sun, 25 May 2014 13:50:35 -0400

mantaray (1.3.82-ubuntu0) precise; urgency=low

  * Corrected Launching

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Sat, 24 May 2014 18:36:05 -0400

mantaray (1.3.81-ubuntu0) precise; urgency=low

  * Updates to MantaRay launcher

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Thu, 22 May 2014 18:08:49 -0400

mantaray (1.3.80-ubuntu0) precise; urgency=low

  * MantaRay updates

 -- root <root@ubuntu>  Sun, 18 May 2014 18:00:10 -0400

mantaray (1.3.8-ubuntu0) precise; urgency=low

  * Bug Fixes in Directory/L01 source types

 -- Chapin Bryce <sansforensics@ubuntu>  Mon, 07 Apr 2014 08:26:18 -0400

mantaray (1.3.7) precise; urgency=low

  * Update to plsit dependencies

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Sat, 05 Apr 2014 15:13:24 -0400

mantaray (1.3.6-ubuntu0) precise; urgency=low

  * Updates to dependencies

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Sat, 05 Apr 2014 12:16:48 -0400

mantaray (1.3.5-ubuntu1) precise; urgency=low

  * Modification to install

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Fri, 04 Apr 2014 15:36:32 -0400

mantaray (1.3.4-ubuntu0) precise; urgency=low

  * Modification to install

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Fri, 04 Apr 2014 12:24:11 -0400

mantaray (1.3.3) precise; urgency=low

  * Version change for code change

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Fri, 04 Apr 2014 11:40:06 -0400

mantaray (1.3.2-dev1) precise; urgency=low

  * edits to install file and file names

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Fri, 04 Apr 2014 11:36:19 -0400

mantaray (1.3.2-dev0) precise; urgency=low

  * Version change for code change

 -- Chapin Bryce <sansforensics@ubuntu>  Fri, 04 Apr 2014 11:13:01 -0400

mantaray (1.3.1-dev1) precise; urgency=low

  * Code updates and bug fixes. See github wiki

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Fri, 04 Apr 2014 11:06:18 -0400

mantaray (1.3.1-dev0) precise; urgency=low

  * Stability Improvements

 -- Chapin Bryce <sansforensics@ubuntu>  Thu, 27 Mar 2014 11:35:31 -0400

mantaray (1.3.0-dev1ubuntu3) precise; urgency=low

  * Additional Fixes for dev version

 -- Chapin Bryce <sansforensics@ubuntu>  Thu, 27 Mar 2014 11:08:46 -0400

mantaray (1.3.0-dev1ubuntu2) precise; urgency=low

  * Fixed additional typos

 -- Chapin Bryce <sansforensics@ubuntu>  Thu, 27 Mar 2014 11:05:01 -0400

mantaray (1.3.0-dev1ubuntu1) precise; urgency=low

  * Fixed vmware glitch caused typo 

 -- Chapin Bryce <sansforensics@ubuntu>  Thu, 27 Mar 2014 09:37:05 -0400

mantaray (1.3.0-dev1) precise; urgency=low

  * Updated parser_timeline and timeline GUI. Added update script

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Wed, 26 Mar 2014 20:07:00 -0400

mantaray (1.3.0-dev0) precise; urgency=low

  * New features include new module File Name Search, updates to volatility, plaso, and bulk extractor. Added launcher to desktop

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Wed, 26 Mar 2014 11:26:00 -0400

mantaray (1.2.9-ubuntu0) precise; urgency=low

  * Fixed bug in Super Timeline when no timezone was selected 

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Wed, 26 Feb 2014 07:35:00 -0500

mantaray (1.2.8-ubuntu9) precise; urgency=low

  * Added windows-perl dependecy and fixed bugs in jumplist and usrclass

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Tue, 25 Feb 2014 13:37:56 -0500

mantaray (1.2.8-ubuntu8) precise; urgency=low

  * Updated Paths and dependencies. Fixed some rr Plugins, vol bin, and plists

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Tue, 25 Feb 2014 09:46:45 -0500

mantaray (1.2.8-ubuntu7) precise; urgency=low

  * Updated paths in code

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Thu, 20 Feb 2014 21:01:10 -0500

mantaray (1.2.8-ubuntu6) precise; urgency=low

  * More corrections...

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Wed, 19 Feb 2014 14:07:35 -0500

mantaray (1.2.8-ubuntu5) precise; urgency=low

  * Correcting Dependencies 

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Wed, 19 Feb 2014 13:30:33 -0500

mantaray (1.2.8-ubuntu4) precise; urgency=low

  * Added Dependencies

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Wed, 19 Feb 2014 13:03:02 -0500

mantaray (1.2.8-ubuntu3) precise; urgency=low

  * Fixed issue with binary

 -- Chapin Bryce <webmaster@mantarayforensics.com>  Sun, 09 Feb 2014 09:24:21 -0500


bootstrap.sh
#!/bin/sh -
#===============================================================================
# vim: softtabstop=4 shiftwidth=4 expandtab fenc=utf-8 spell spelllang=en cc=81
#===============================================================================


#--- FUNCTION ----------------------------------------------------------------
# NAME: __function_defined
# DESCRIPTION: Checks if a function is defined within this scripts scope
# PARAMETERS: function name
# RETURNS: 0 or 1 as in defined or not defined
#-------------------------------------------------------------------------------
__function_defined() {
    FUNC_NAME=$1
    if [ "$(command -v $FUNC_NAME)x" != "x" ]; then
        echoinfo "Found function $FUNC_NAME"
        return 0
    fi
    
    echodebug "$FUNC_NAME not found...."
    return 1
}

#--- FUNCTION ----------------------------------------------------------------
# NAME: __strip_duplicates
# DESCRIPTION: Strip duplicate strings
#-------------------------------------------------------------------------------
__strip_duplicates() {
    echo $@ | tr -s '[:space:]' '\n' | awk '!x[$0]++'
}

#--- FUNCTION ----------------------------------------------------------------
# NAME: echoerr
# DESCRIPTION: Echo errors to stderr.
#-------------------------------------------------------------------------------
echoerror() {
    printf "${RC} * ERROR${EC}: $@\n" 1>&2;
}

#--- FUNCTION ----------------------------------------------------------------
# NAME: echoinfo
# DESCRIPTION: Echo information to stdout.
#-------------------------------------------------------------------------------
echoinfo() {
    printf "${GC} * INFO${EC}: %s\n" "$@";
}

#--- FUNCTION ----------------------------------------------------------------
# NAME: echowarn
# DESCRIPTION: Echo warning informations to stdout.
#-------------------------------------------------------------------------------
echowarn() {
    printf "${YC} * WARN${EC}: %s\n" "$@";
}

#--- FUNCTION ----------------------------------------------------------------
# NAME: echodebug
# DESCRIPTION: Echo debug information to stdout.
#-------------------------------------------------------------------------------
echodebug() {
    if [ $_ECHO_DEBUG -eq $BS_TRUE ]; then
        printf "${BC} * DEBUG${EC}: %s\n" "$@";
    fi
}

#---  FUNCTION  ----------------------------------------------------------------
#          NAME:  __apt_get_install_noinput
#   DESCRIPTION:  (DRY) apt-get install with noinput options
#-------------------------------------------------------------------------------
__apt_get_install_noinput() {
    apt-get install -y -o DPkg::Options::=--force-confold $@; return $?
}

#---  FUNCTION  ----------------------------------------------------------------
#          NAME:  __apt_get_upgrade_noinput
#   DESCRIPTION:  (DRY) apt-get upgrade with noinput options
#-------------------------------------------------------------------------------
__apt_get_upgrade_noinput() {
    apt-get upgrade -y -o DPkg::Options::=--force-confold $@; return $?
}

#---  FUNCTION  ----------------------------------------------------------------
#          NAME:  __pip_install_noinput
#   DESCRIPTION:  (DRY)
#-------------------------------------------------------------------------------
__pip_install_noinput() {
    pip install --upgrade $@; return $?
}

__ScriptVersion=20140505	

__enable_universe_repository() {
    if [ "x$(grep -R universe /etc/apt/sources.list /etc/apt/sources.list.d/ | grep -v '#')" != "x" ]; then
        # The universe repository is already enabled
        return 0
    fi

    echodebug "Enabling the universe repository"

    # Ubuntu versions higher than 12.04 do not live in the old repositories
    if [ $DISTRO_MAJOR_VERSION -gt 12 ] || ([ $DISTRO_MAJOR_VERSION -eq 12 ] && [ $DISTRO_MINOR_VERSION -gt 04 ]); then
        add-apt-repository -y "deb http://archive.ubuntu.com/ubuntu $(lsb_release -sc) universe" || return 1
    elif [ $DISTRO_MAJOR_VERSION -lt 11 ] && [ $DISTRO_MINOR_VERSION -lt 10 ]; then
        # Below Ubuntu 11.10, the -y flag to add-apt-repository is not supported
        add-apt-repository "deb http://old-releases.ubuntu.com/ubuntu $(lsb_release -sc) universe" || return 1
    fi

    add-apt-repository -y "deb http://old-releases.ubuntu.com/ubuntu $(lsb_release -sc) universe" || return 1

    return 0
}

__check_unparsed_options() {
    shellopts="$1"
    # grep alternative for SunOS
    if [ -f /usr/xpg4/bin/grep ]; then
        grep='/usr/xpg4/bin/grep'
    else
        grep='grep'
    fi
    unparsed_options=$( echo "$shellopts" | ${grep} -E '(^|[[:space:]])[-]+[[:alnum:]]' )
    if [ "x$unparsed_options" != "x" ]; then
        usage
        echo
        echoerror "options are only allowed before install arguments"
        echo
        exit 1
    fi
}

configure_cpan() {
    (echo y;echo o conf prerequisites_policy follow;echo o conf commit)|cpan > /dev/null
}

usage() {
    echo "usage"
    echo
    echo "h - print this message"
    echo "s - install the skin"
    echo "i - install the tools"
    echo "c - configure only"
    echo "y - yes to all"
    exit 1
}

install_ubuntu_deps() {
    apt-get update

    __apt_get_install_noinput python-software-properties || return 1

    __enable_universe_repository || return 1

    add-apt-repository -y ppa:sift/$@ || return 1

    add-apt-repository -y ppa:mantaray/$@ || return 1

    apt-get update

    __apt_get_upgrade_noinput || return 1

    return 0
}

install_ubuntu() {
    packages="mantaray binplist bulk-extractor dconf-tools dos2unix libevt libevt-dev libevt-tools libevt-python libevt libevtx-dev libevtx-tools libevtx-python libewf libewf-dev libewf-tools libewf-python libfvde libfvde-dev libfvde-tools liblightgrep libolecf libolecf-dev libolecf-tools libolecf-python libolecf-tools libplist++1 libplist-dev libplist1 libregf libregf-python libregf-tools libregf-dev libvshadow libvshadow-tools libvshadow-dev libvshadow-python log2timeline-perl python-plaso pytsk3 regripper sift sift-scripts g++ gcc libtsk libtsk-dev sleuthkit windows-perl python-setuptools python-pip python3 git"

    if [ "$@" = "dev" ]; then
        packages="$packages"
    elif [ "$@" = "stable" ]; then
        packages="$packages"
    fi

    __apt_get_install_noinput $packages || return 1

    return 0
}

install_pip_packages() {
    pip_packages="rekall docopt python-evtx python-registry"

    sudo easy_install pip
    sudo pip install --upgrade pip virtualenv virtualenvwrapper

    if [ "$@" = "dev" ]; then
        pip_packages="$pip_packages"
    elif [ "$@" = "stable" ]; then
        pip_packages="$pip_packages"
    fi

    __pip_install_noinput $pip_packages || return 1

    return 0
}

install_perl_modules() {
	# Required by macl.pl script
	perl -MCPAN -e "install Net::Wigle" > /dev/null
}

configure_ubuntu() {
	if [ ! -d /cases ]; then
		mkdir -p /cases
		chown $SUDO_USER:$SUDO_USER /cases
		chmod 775 /cases
		chmod g+s /cases
	fi

	for dir in usb vss shadow windows_mount e01 aff ewf bde iscsi
	do
		if [ ! -d /mnt/$dir ]; then
			mkdir -p /mnt/$dir
		fi
	done

	for NUM in 1 2 3 4 5
	do
		if [ ! -d /mnt/windows_mount$NUM ]; then
			mkdir -p /mnt/windows_mount$NUM
		fi
		if [ ! -d /mnt/ewf$NUM ]; then
			mkdir -p /mnt/ewf$NUM
		fi
	done
 
	for NUM in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
	do
		if [ ! -d /mnt/shadow/vss$NUM ]; then
			mkdir -p /mnt/shadow/vss$NUM
		fi
		if [ ! -d /mnt/shadow_mount/vss$NUM ]; then
			mkdir -p /mnt/shadow_mount/vss$NUM
		fi
	done
	
	if [ ! -L /usr/bin/vol.py ]; then
		ln -s /usr/bin/vol /usr/bin/vol.py
	fi
	if [ ! -L /usr/bin/log2timeline ]; then
		ln -s /usr/bin/log2timeline_legacy /usr/bin/log2timeline
	fi
	if [ ! -L /usr/bin/kedit ]; then
		ln -s /usr/bin/gedit /usr/bin/kedit
	fi
	if [ ! -L /usr/bin/mount_ewf.py ] && [ ! -e /usr/bin/mount_ewf.py ]; then
		ln -s /usr/bin/ewfmount /usr/bin/mount_ewf.py
	fi
}

configure_ubuntu_skin() {
	if [ ! -d /home/$SUDO_USER/.config/autostart ]; then
		sudo -u $SUDO_USER mkdir -p /home/$SUDO_USER/.config/autostart
	fi

	sudo -u $SUDO_USER gsettings set org.gnome.desktop.background picture-uri file:///usr/share/mantaray/images/Mantaray_Logo_Template_Full_Screen.gif
	sudo -u $SUDO_USER dconf write /desktop/unity/launcher/favorites "['nautilus.desktop', 'gnome-terminal.desktop', 'firefox.desktop', 'MantaRay.desktop', 'gnome-screenshot.desktop', 'gcalctool.desktop', 'bless.desktop', 'dff.desktop', 'autopsy.desktop', 'wireshark.desktop']"

	if [ ! -L /home/$SUDO_USER/Desktop/cases ]; then
		sudo -u $SUDO_USER ln -s /cases /home/$SUDO_USER/Desktop/cases
		cp /usr/share/mantaray/MantaRay.Desktop /home/$SUDO_USER/Desktop
	fi
  
	if [ ! -L /home/$SUDO_USER/Desktop/mount_points ]; then
		sudo -u $SUDO_USER ln -s /mnt /home/$SUDO_USER/Desktop/mount_points
	fi

	# Clean up broken symlinks
	find -L /home/$SUDO_USER/Desktop -type l -delete

	for file in /usr/share/sift/resources/*.pdf
	do
		base=`basename $file`
		if [ ! -L /home/$SUDO_USER/Desktop/$base ]; then
			sudo -u $SUDO_USER ln -s $file /home/$SUDO_USER/Desktop/$base
		fi
	done
	
	if [ ! -L /home/$SUDO_USER/.config/autostart ]; then
		sudo -u $SUDO_USER cp /usr/share/sift/other/gnome-terminal.desktop /home/$SUDO_USER/.config/autostart
	fi

	gsettings set com.canonical.unity-greeter background file:///usr/share/mantaray/images/Mantaray_Logo_Template_Full_Screen.gif

	# Checkout code from sift-files and put these files into place
	CDIR=$(pwd)
	git clone https://github.com/sans-dfir/sift-files /tmp/sift-files
	cd /tmp/sift-files
	bash install.sh
	cd $CDIR
	rm -r -f /tmp/sift-files

	# Make sure we replace the SIFT_USER template with our actual
	# user so there is write permissions to samba.
	sed -i "s/SIFT_USER/$SUDO_USER/g" /etc/samba/smb.conf

	# Restart samba services 
	service smbd restart
	service nmbd restart

	# Disable services
	update-rc.d tor disable

	# Make sure to remove all ^M from regripper plugins
	# Not sure why they are there in the first place ...
	dos2unix -ascii /usr/share/regripper/*

	OLD_HOSTNAME=$(hostname)
	sed -i "s/$OLD_HOSTNAME/mantarayforensics/g" /etc/hosts
	echo "mantarayforensics" > /etc/hostname
	hostname mantarayforensics

	if ! grep -i "set -o noclobber" $HOME/.bashrc > /dev/null 2>&1
	then
		echo "set -o noclobber" >> $HOME/.bashrc
	fi

	if ! grep -i "alias mountwin" $HOME/.bash_aliases > /dev/null 2>&1
	then
		echo "alias mountwin='mount -o ro,loop,show_sys_files,streams_interface=windows'" >> $HOME/.bash_aliases
	fi
}


complete_message() {
    echo
    echo "Installation Completed!"
    echo 
    echo "Formal documentation is in progress."
    echo
    echo "http://mantaray.readthedocs.org"
    echo
}

complete_message_skin() {
    echo "The hostname was changed, you should relogin or reboot for it to take full effect."
    echo
    echo "sudo reboot"
    echo
}

CONFIGURE_ONLY=0
SKIN=0
INSTALL=1
YESTOALL=0

OS=$(lsb_release -si)
ARCH=$(uname -m | sed 's/x86_//;s/i[3-6]86/32/')
VER=$(lsb_release -sr)


if [ $OS != "Ubuntu" ]; then
    echo "MantaRay is only installable on Ubuntu operating systems at this time."
    exit 1
fi

if [ $ARCH != "64" ]; then
    echo "MantaRay is only installable on a 64 bit architecture at this time."
    exit 2
fi

if [ $VER != "12.04" ]; then
    echo "MantaRay is only installable on Ubuntu 12.04 at this time."
    exit 3
fi

if [ `whoami` != "root" ]; then
    echo "MantaRay Bootstrap must be run as root!"
    exit 3
fi

if [ "$SUDO_USER" = "" ]; then
    echo "The SUDO_USER variable doesn't seem to be set"
    exit 4
fi


while getopts ":hvcsiy" opt
do
case "${opt}" in
    h ) usage; exit 1;;
    v ) echo "$0 -- Version $__ScriptVersion"; exit 0 ;;
    s ) SKIN=1 ;;
    i ) INSTALL=1 ;;
    c ) CONFIGURE_ONLY=1; INSTALL=0; SKIN=0; ;;
    y ) YESTOALL=1 ;;
    \?) echo
        echoerror "Option does not exist: $OPTARG"
        usage
        exit 1
        ;;
esac
done

shift $(($OPTIND-1))

if [ "$#" -eq 0 ]; then
    ITYPE="stable"
else
    __check_unparsed_options "$*"
    ITYPE=$1
    shift
fi

# Check installation type
if [ "$(echo $ITYPE | egrep '(dev|stable)')x" = "x" ]; then
    echoerror "Installation type \"$ITYPE\" is not known..."
    exit 1
fi


echo "Welcome to the MantaRay Bootstrap"
echo "This script will now proceed to configure your system."

if [ "$YESTOALL" -eq 1 ]; then
    echo "You supplied the -y option, this script will not exit for any reason"
fi

if [ "$SKIN" -eq 1 ] && [ "$YESTOALL" -eq 0 ]; then
    echo
    echo "You have chosen to apply the MantaRay skin to your ubuntu system."
    echo 
    echo "You did not choose to say YES to all, so we are going to exit."
    echo
    echo "Your current user is: $SUDO_USER"
    echo
    echo "Re-run this command with the -y option"
    echo
    exit 10
fi

if [ "$INSTALL" -eq 1 ] && [ "$CONFIGURE_ONLY" -eq 0 ]; then
    install_ubuntu_deps $ITYPE
    install_ubuntu $ITYPE
    install_pip_packages $ITYPE
    configure_cpan
    install_perl_modules
fi

configure_ubuntu

if [ "$KIBANA" -eq 1 ]; then
    echo "Kibana is not supported in this version, Please see the Kibana branch of MantaRay for the supporting development code"
fi

if [ "$SKIN" -eq 1 ]; then
    configure_ubuntu_skin
fi

complete_message

if [ "$SKIN" -eq 1 ]; then
    complete_message_skin
fi


